milesightct101/sensor_test.go

```diff
@@ -16,7 +25,255 @@ const (
 	testGatewayName = "gateway"
 )
 
-var testInterval = 5.0
+var (
+	testNodeReadings = map[string]interface{}{"reading": 1}
+	testInterval     = 5.0
+)
+
+func createMockGateway() *inject.Sensor {
+	mockGateway := &inject.Sensor{}
+	mockGateway.DoFunc = func(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+		if _, ok := cmd["validate"]; ok {
+			return map[string]interface{}{"validate": 1.0}, nil
+		}
+		if _, ok := cmd[node.GatewaySendDownlinkKey]; ok {
+			return map[string]interface{}{node.GatewaySendDownlinkKey: "downlink added"}, nil
+		}
+		return map[string]interface{}{}, nil
+	}
+	mockGateway.ReadingsFunc = func(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
+		readings := make(map[string]interface{})
+		readings["test-ct101"] = testNodeReadings
+		return readings, nil
+	}
+	return mockGateway
+}
```

Comment:
using this in every sensor. Didn't seem worth to figure out the right way to make the helper, and making a test/helper thats too complicated isn't helpful anymore